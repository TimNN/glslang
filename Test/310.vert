#version 310 es

shared vec4 s;   // ERROR
layout(local_size_x = 2) out;  // ERROR
buffer vec4 v;  // ERROR

layout(location = 2) uniform mat4 x;
layout(location = 3) uniform mat4 y;
layout(location = 2) out mat4 xi;
layout(location = 3) out mat4 yi;  // ERROR, locations conflict with xi

void main()
{
    uvec2 u2;
    u2 = uaddCarry(u2, u2, u2);
    uint u1;
    u1 = usubBorrow(u1, u1, u1);
    uvec4 u4;
    umulExtended(u4, u4, u4, u4);
    ivec4 i4;
    imulExtended(i4, i4, i4, i4);
    int i1;
    i1 = bitfieldExtract(i1, 4, 5);
    uvec3 u3;
    u3 = bitfieldExtract(u3, 4, 5);
    ivec3 i3;
    i3 = bitfieldInsert(i3, i3, 4, 5);
    u1 = bitfieldInsert(u1, u1, 4, 5);
    ivec2 i2;
    i2 = bitfieldReverse(i2);
    u4 = bitfieldReverse(u4);
    i1 = bitCount(i1);
    i3 = bitCount(u3);
    i2 = findLSB(i2);
    i4 = findLSB(u4);
    i1 = findMSB(i1);
    i2 = findMSB(u2);

    vec3 v3;
    v3 = frexp(v3, i3);
    vec2 v2;
    v2 = ldexp(v2, i2);

    mediump vec4 v4;
    u1 = packUnorm4x8(v4);
    u1 = packSnorm4x8(v4);
    v4 = unpackUnorm4x8(u1);
    v4 = unpackSnorm4x8(u1);
}
