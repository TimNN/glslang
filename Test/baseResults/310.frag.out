310.frag
Warning, version 310 is not yet complete; most version-specific features are present, but some are missing.
ERROR: 0:8: 'origin_upper_left' : not supported with this profile: es
ERROR: 0:8: 'pixel_center_integer' : not supported with this profile: es
ERROR: 0:8: 'gl_FragCoord' : identifiers starting with "gl_" are reserved 
ERROR: 0:11: 'location' : overlapping use of location 2
ERROR: 0:15: 'usampler2DRect' : Reserved word. 
ERROR: 0:15: 'sampler/image' : type requires declaration of default precision qualifier 
ERROR: 0:23: 'variable indexing sampler array' : not supported with this profile: es
ERROR: 0:27: 'textureGatherOffsets' : no matching overloaded function found 
ERROR: 0:27: '=' :  cannot convert from 'const float' to 'mediump 4-component vector of uint'
ERROR: 0:30: 'textureGatherOffset(...)' : must be a compile-time constant: component argument
ERROR: 0:31: 'textureGatherOffset(...)' : must be 0, 1, 2, or 3: component argument
ERROR: 0:42: 'texel offset' : argument must be compile-time constant 
ERROR: 0:44: 'texel offset' : value is out of range: [gl_MinProgramTexelOffset, gl_MaxProgramTexelOffset]
ERROR: 0:44: 'texel offset' : value is out of range: [gl_MinProgramTexelOffset, gl_MaxProgramTexelOffset]
ERROR: 0:45: 'texel offset' : value is out of range: [gl_MinProgramTexelOffset, gl_MaxProgramTexelOffset]
ERROR: 0:45: 'texel offset' : value is out of range: [gl_MinProgramTexelOffset, gl_MaxProgramTexelOffset]
ERROR: 16 compilation errors.  No code generated.


Shader version: 310
ERROR: node is still EOpNull!
0:21  Function Definition: main( (void)
0:21    Function Parameters: 
0:23    Sequence
0:23      Sequence
0:23        move second child to first child (highp 4-component vector of float)
0:23          'v' (mediump 4-component vector of float)
0:23          Function Call: texture(s21;vf2; (highp 4-component vector of float)
0:23            indirect index (highp sampler2D)
0:23              'arrayedSampler' (uniform 5-element array of highp sampler2D)
0:23              'i' (uniform mediump int)
0:23            'c2D' (smooth in mediump 2-component vector of float)
0:28      Sequence
0:28        move second child to first child (highp 4-component vector of float)
0:28          'v4' (mediump 4-component vector of float)
0:28          Function Call: textureGather(s21;vf2; (highp 4-component vector of float)
0:28            direct index (highp sampler2D)
0:28              'arrayedSampler' (uniform 5-element array of highp sampler2D)
0:28              Constant:
0:28                0 (const int)
0:28            'c2D' (smooth in mediump 2-component vector of float)
0:29      Sequence
0:29        move second child to first child (mediump 4-component vector of int)
0:29          'iv4' (mediump 4-component vector of int)
0:29          Function Call: textureGatherOffset(isA21;vf3;vi2;i1; (mediump 4-component vector of int)
0:29            'isamp2DA' (uniform highp isampler2DArray)
0:29            Constant:
0:29              0.100000
0:29              0.100000
0:29              0.100000
0:29            Constant:
0:29              1 (const int)
0:29              1 (const int)
0:29            Constant:
0:29              3 (const int)
0:30      move second child to first child (mediump 4-component vector of int)
0:30        'iv4' (mediump 4-component vector of int)
0:30        Function Call: textureGatherOffset(isA21;vf3;vi2;i1; (mediump 4-component vector of int)
0:30          'isamp2DA' (uniform highp isampler2DArray)
0:30          Constant:
0:30            0.100000
0:30            0.100000
0:30            0.100000
0:30          Constant:
0:30            1 (const int)
0:30            1 (const int)
0:30          'i' (uniform mediump int)
0:31      move second child to first child (mediump 4-component vector of int)
0:31        'iv4' (mediump 4-component vector of int)
0:31        Function Call: textureGatherOffset(isA21;vf3;vi2;i1; (mediump 4-component vector of int)
0:31          'isamp2DA' (uniform highp isampler2DArray)
0:31          Constant:
0:31            0.100000
0:31            0.100000
0:31            0.100000
0:31          Constant:
0:31            1 (const int)
0:31            1 (const int)
0:31          Constant:
0:31            4 (const int)
0:32      move second child to first child (mediump 4-component vector of int)
0:32        'iv4' (mediump 4-component vector of int)
0:32        Function Call: textureGatherOffset(isA21;vf3;vi2;i1; (mediump 4-component vector of int)
0:32          'isamp2DA' (uniform highp isampler2DArray)
0:32          Constant:
0:32            0.100000
0:32            0.100000
0:32            0.100000
0:32          Constant:
0:32            1 (const int)
0:32            1 (const int)
0:32          Constant:
0:32            3 (const int)
0:33      move second child to first child (mediump 4-component vector of int)
0:33        'iv4' (mediump 4-component vector of int)
0:33        Function Call: textureGatherOffset(isA21;vf3;vi2; (mediump 4-component vector of int)
0:33          'isamp2DA' (uniform highp isampler2DArray)
0:33          Constant:
0:33            0.100000
0:33            0.100000
0:33            0.100000
0:33          Construct ivec2 (2-component vector of int)
0:33            'i' (uniform mediump int)
0:38  Function Definition: foo23( (void)
0:38    Function Parameters: 
0:?     Sequence
0:42      Function Call: textureProjGradOffset(us21;vf4;vf2;vf2;vi2; (mediump 4-component vector of uint)
0:42        'usamp2d' (uniform highp usampler2D)
0:42        'outp' (out mediump 4-component vector of float)
0:42        Constant:
0:42          0.000000
0:42          0.000000
0:42        Constant:
0:42          0.000000
0:42          0.000000
0:42        Convert float to int (2-component vector of int)
0:42          'c2D' (smooth in mediump 2-component vector of float)
0:43      Function Call: textureProjGradOffset(us21;vf4;vf2;vf2;vi2; (mediump 4-component vector of uint)
0:43        'usamp2d' (uniform highp usampler2D)
0:43        'outp' (out mediump 4-component vector of float)
0:43        Constant:
0:43          0.000000
0:43          0.000000
0:43        Constant:
0:43          0.000000
0:43          0.000000
0:43        Constant:
0:43          3 (const int)
0:43          4 (const int)
0:44      Function Call: textureProjGradOffset(us21;vf4;vf2;vf2;vi2; (mediump 4-component vector of uint)
0:44        'usamp2d' (uniform highp usampler2D)
0:44        'outp' (out mediump 4-component vector of float)
0:44        Constant:
0:44          0.000000
0:44          0.000000
0:44        Constant:
0:44          0.000000
0:44          0.000000
0:44        Constant:
0:44          15 (const int)
0:44          16 (const int)
0:45      Function Call: textureProjGradOffset(us21;vf4;vf2;vf2;vi2; (mediump 4-component vector of uint)
0:45        'usamp2d' (uniform highp usampler2D)
0:45        'outp' (out mediump 4-component vector of float)
0:45        Constant:
0:45          0.000000
0:45          0.000000
0:45        Constant:
0:45          0.000000
0:45          0.000000
0:45        Constant:
0:45          -10 (const int)
0:45          20 (const int)
0:47      Test condition and select (void)
0:47        Condition
0:47        'gl_HelperInvocation' (in bool)
0:47        true case
0:48        Pre-Increment (mediump 4-component vector of float)
0:48          'outp' (out mediump 4-component vector of float)
0:50      Sequence
0:50        move second child to first child (mediump int)
0:50          'sum' (mediump int)
0:50          Constant:
0:50            32 (const int)
0:58      move second child to first child (bool)
0:58        'b1' (bool)
0:58        mix (bool)
0:58          'b2' (bool)
0:58          'b3' (bool)
0:58          'b' (bool)
0:59      Sequence
0:59        move second child to first child (mediump 3-component vector of uint)
0:59          'um3' (mediump 3-component vector of uint)
0:59          mix (mediump 3-component vector of uint)
0:59            Construct uvec3 (mediump 3-component vector of uint)
0:59              Convert int to uint (mediump uint)
0:59                'i' (uniform mediump int)
0:59            Construct uvec3 (mediump 3-component vector of uint)
0:59              Convert int to uint (mediump uint)
0:59                'i' (uniform mediump int)
0:59            Construct bvec3 (3-component vector of bool)
0:59              'b' (bool)
0:60      Sequence
0:60        move second child to first child (mediump 4-component vector of int)
0:60          'im4' (mediump 4-component vector of int)
0:60          mix (mediump 4-component vector of int)
0:60            Construct ivec4 (mediump 4-component vector of int)
0:60              'i' (uniform mediump int)
0:60            Construct ivec4 (mediump 4-component vector of int)
0:60              'i' (uniform mediump int)
0:60            Construct bvec4 (4-component vector of bool)
0:60              'b' (bool)
0:?   Linker Objects
0:?     'gl_FragCoord' (smooth in mediump 4-component vector of float)
0:?     'v3' (layout(location=2 ) smooth in mediump 3-component vector of float)
0:?     'yi' (layout(location=2 ) smooth in mediump 4X4 matrix of float)
0:?     'arrayedSampler' (uniform 5-element array of highp sampler2D)
0:?     'usamp2d' (uniform highp usampler2D)
0:?     'samp2dr' (uniform mediump usampler2DRect)
0:?     'isamp2DA' (uniform highp isampler2DArray)
0:?     'c2D' (smooth in mediump 2-component vector of float)
0:?     'i' (uniform mediump int)
0:?     'outp' (out mediump 4-component vector of float)


Linked fragment stage:


Shader version: 310
ERROR: node is still EOpNull!
0:21  Function Definition: main( (void)
0:21    Function Parameters: 
0:23    Sequence
0:23      Sequence
0:23        move second child to first child (highp 4-component vector of float)
0:23          'v' (mediump 4-component vector of float)
0:23          Function Call: texture(s21;vf2; (highp 4-component vector of float)
0:23            indirect index (highp sampler2D)
0:23              'arrayedSampler' (uniform 5-element array of highp sampler2D)
0:23              'i' (uniform mediump int)
0:23            'c2D' (smooth in mediump 2-component vector of float)
0:28      Sequence
0:28        move second child to first child (highp 4-component vector of float)
0:28          'v4' (mediump 4-component vector of float)
0:28          Function Call: textureGather(s21;vf2; (highp 4-component vector of float)
0:28            direct index (highp sampler2D)
0:28              'arrayedSampler' (uniform 5-element array of highp sampler2D)
0:28              Constant:
0:28                0 (const int)
0:28            'c2D' (smooth in mediump 2-component vector of float)
0:29      Sequence
0:29        move second child to first child (mediump 4-component vector of int)
0:29          'iv4' (mediump 4-component vector of int)
0:29          Function Call: textureGatherOffset(isA21;vf3;vi2;i1; (mediump 4-component vector of int)
0:29            'isamp2DA' (uniform highp isampler2DArray)
0:29            Constant:
0:29              0.100000
0:29              0.100000
0:29              0.100000
0:29            Constant:
0:29              1 (const int)
0:29              1 (const int)
0:29            Constant:
0:29              3 (const int)
0:30      move second child to first child (mediump 4-component vector of int)
0:30        'iv4' (mediump 4-component vector of int)
0:30        Function Call: textureGatherOffset(isA21;vf3;vi2;i1; (mediump 4-component vector of int)
0:30          'isamp2DA' (uniform highp isampler2DArray)
0:30          Constant:
0:30            0.100000
0:30            0.100000
0:30            0.100000
0:30          Constant:
0:30            1 (const int)
0:30            1 (const int)
0:30          'i' (uniform mediump int)
0:31      move second child to first child (mediump 4-component vector of int)
0:31        'iv4' (mediump 4-component vector of int)
0:31        Function Call: textureGatherOffset(isA21;vf3;vi2;i1; (mediump 4-component vector of int)
0:31          'isamp2DA' (uniform highp isampler2DArray)
0:31          Constant:
0:31            0.100000
0:31            0.100000
0:31            0.100000
0:31          Constant:
0:31            1 (const int)
0:31            1 (const int)
0:31          Constant:
0:31            4 (const int)
0:32      move second child to first child (mediump 4-component vector of int)
0:32        'iv4' (mediump 4-component vector of int)
0:32        Function Call: textureGatherOffset(isA21;vf3;vi2;i1; (mediump 4-component vector of int)
0:32          'isamp2DA' (uniform highp isampler2DArray)
0:32          Constant:
0:32            0.100000
0:32            0.100000
0:32            0.100000
0:32          Constant:
0:32            1 (const int)
0:32            1 (const int)
0:32          Constant:
0:32            3 (const int)
0:33      move second child to first child (mediump 4-component vector of int)
0:33        'iv4' (mediump 4-component vector of int)
0:33        Function Call: textureGatherOffset(isA21;vf3;vi2; (mediump 4-component vector of int)
0:33          'isamp2DA' (uniform highp isampler2DArray)
0:33          Constant:
0:33            0.100000
0:33            0.100000
0:33            0.100000
0:33          Construct ivec2 (2-component vector of int)
0:33            'i' (uniform mediump int)
0:38  Function Definition: foo23( (void)
0:38    Function Parameters: 
0:?     Sequence
0:42      Function Call: textureProjGradOffset(us21;vf4;vf2;vf2;vi2; (mediump 4-component vector of uint)
0:42        'usamp2d' (uniform highp usampler2D)
0:42        'outp' (out mediump 4-component vector of float)
0:42        Constant:
0:42          0.000000
0:42          0.000000
0:42        Constant:
0:42          0.000000
0:42          0.000000
0:42        Convert float to int (2-component vector of int)
0:42          'c2D' (smooth in mediump 2-component vector of float)
0:43      Function Call: textureProjGradOffset(us21;vf4;vf2;vf2;vi2; (mediump 4-component vector of uint)
0:43        'usamp2d' (uniform highp usampler2D)
0:43        'outp' (out mediump 4-component vector of float)
0:43        Constant:
0:43          0.000000
0:43          0.000000
0:43        Constant:
0:43          0.000000
0:43          0.000000
0:43        Constant:
0:43          3 (const int)
0:43          4 (const int)
0:44      Function Call: textureProjGradOffset(us21;vf4;vf2;vf2;vi2; (mediump 4-component vector of uint)
0:44        'usamp2d' (uniform highp usampler2D)
0:44        'outp' (out mediump 4-component vector of float)
0:44        Constant:
0:44          0.000000
0:44          0.000000
0:44        Constant:
0:44          0.000000
0:44          0.000000
0:44        Constant:
0:44          15 (const int)
0:44          16 (const int)
0:45      Function Call: textureProjGradOffset(us21;vf4;vf2;vf2;vi2; (mediump 4-component vector of uint)
0:45        'usamp2d' (uniform highp usampler2D)
0:45        'outp' (out mediump 4-component vector of float)
0:45        Constant:
0:45          0.000000
0:45          0.000000
0:45        Constant:
0:45          0.000000
0:45          0.000000
0:45        Constant:
0:45          -10 (const int)
0:45          20 (const int)
0:47      Test condition and select (void)
0:47        Condition
0:47        'gl_HelperInvocation' (in bool)
0:47        true case
0:48        Pre-Increment (mediump 4-component vector of float)
0:48          'outp' (out mediump 4-component vector of float)
0:50      Sequence
0:50        move second child to first child (mediump int)
0:50          'sum' (mediump int)
0:50          Constant:
0:50            32 (const int)
0:58      move second child to first child (bool)
0:58        'b1' (bool)
0:58        mix (bool)
0:58          'b2' (bool)
0:58          'b3' (bool)
0:58          'b' (bool)
0:59      Sequence
0:59        move second child to first child (mediump 3-component vector of uint)
0:59          'um3' (mediump 3-component vector of uint)
0:59          mix (mediump 3-component vector of uint)
0:59            Construct uvec3 (mediump 3-component vector of uint)
0:59              Convert int to uint (mediump uint)
0:59                'i' (uniform mediump int)
0:59            Construct uvec3 (mediump 3-component vector of uint)
0:59              Convert int to uint (mediump uint)
0:59                'i' (uniform mediump int)
0:59            Construct bvec3 (3-component vector of bool)
0:59              'b' (bool)
0:60      Sequence
0:60        move second child to first child (mediump 4-component vector of int)
0:60          'im4' (mediump 4-component vector of int)
0:60          mix (mediump 4-component vector of int)
0:60            Construct ivec4 (mediump 4-component vector of int)
0:60              'i' (uniform mediump int)
0:60            Construct ivec4 (mediump 4-component vector of int)
0:60              'i' (uniform mediump int)
0:60            Construct bvec4 (4-component vector of bool)
0:60              'b' (bool)
0:?   Linker Objects
0:?     'gl_FragCoord' (smooth in mediump 4-component vector of float)
0:?     'v3' (layout(location=2 ) smooth in mediump 3-component vector of float)
0:?     'yi' (layout(location=2 ) smooth in mediump 4X4 matrix of float)
0:?     'arrayedSampler' (uniform 5-element array of highp sampler2D)
0:?     'usamp2d' (uniform highp usampler2D)
0:?     'samp2dr' (uniform mediump usampler2DRect)
0:?     'isamp2DA' (uniform highp isampler2DArray)
0:?     'c2D' (smooth in mediump 2-component vector of float)
0:?     'i' (uniform mediump int)
0:?     'outp' (out mediump 4-component vector of float)

